onnxmltools/convert/sklearn/shape_calculators/LinearClassifier.py

```diff
@@ -19,43 +19,33 @@ def calculate_sklearn_linear_classifier_output_shapes(operator):
     appear in this operator's output list, we should further generate a map storing all classes' probabilities.
 
     Allowed input/output patterns are
-        1. [N, C] ---> [N, 1], A sequence of map
+        1. [N, C] ---> [N], [N, E] (binary-class)
+        2. [N, 1], ..., [N, 1] ---> [N], [N, E] (binary-class)
+        3. [N, C] ---> [N], A sequence of map (multi-class)
+        4. [N, 1], ..., [N, 1] ---> [N], A sequence of map (multi-class)
 
     Note that the second case is not allowed as long as ZipMap only produces dictionary.
     '''
-    check_input_and_output_numbers(operator, input_count_range=1, output_count_range=[1, 2])
+    check_input_and_output_numbers(operator, input_count_range=[1, None], output_count_range=[1, 2])
     check_input_and_output_types(operator, good_input_types=[FloatTensorType, Int64TensorType])
 
     if len(operator.inputs[0].type.shape) != 2:
         raise RuntimeError('Input must be a [N, C]-tensor')
 
+    if len(operator.inputs) > 1 and all((i.type.shape[1] != 1) for i in operator.inputs):
+        raise RuntimeError('Multiple inputs must be a [N, 1]-tensors')
+
     N = operator.inputs[0].type.shape[0]
 
     class_labels = operator.raw_operator.classes_
     if all(isinstance(i, np.ndarray) for i in class_labels):
         class_labels = np.concatenate(class_labels)
     if all(isinstance(i, (six.string_types, six.text_type)) for i in class_labels):
         operator.outputs[0].type = StringTensorType(shape=[N, 1])
-        if len(class_labels) > 2 or operator.type != 'SklearnLinearSVC':
-            # For multi-class classifier, we produce a map for encoding the probabilities of all classes
-            if operator.targeted_onnx_version < StrictVersion('1.2'):
```

Comment:
why is the backward compatibility code deleted?

Reply:
I was asked to convert LightGBM multi-class model also to ONNX for PredictionUnit API in Windows 10. I am not sure why this was written to output SequenceType. Was there any particular reason?

Reply:
it complys with the onnx spec https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.ZipMap

Reply:
Yes, I know ZipMap complies onnx spec. I am asking why do you do ZipMap operation after TreeEnsembleClassifier?

Reply:
It's an idea from Core ML. I personally think it's nice to associate label names and their probabilities from user perspective. On the other hand, it's doable to remove all ZipMap's in scikit-learn conversions but test data should be modified as well.

Reply:
I reached out to Microsoft ONNX team and they said they use ZipMap to be consistent on Tree implementation in scikit-learn (i.e., all tree models return probabilities as map not [N, E]). Following up with them more about how I used this tool for our requirements.